[Bug]: TestSparseInvertedIndexV3Negative setup declares 12 vector fields, exceeding maxVectorFieldNum=10 (all 9 L2 negative cases error at setup)

**Is there an existing issue for this?**

- [x] I have searched the existing issues (no match for `TestSparseInvertedIndexV3Negative` / sparse-v3 negative setup).

### Environment

- Milvus: master (`tests/python_client/milvus_client/`)
- pymilvus: `3.1.0rc8` (per `tests/python_client/requirements.txt`)
- Deployment: standalone, default config (`proxy.maxVectorFieldNum = 10`)

### Current Behavior

`TestSparseInvertedIndexV3Negative` (added in #50105, `tests/python_client/milvus_client/test_milvus_client_sparse_inverted_index.py:955`) builds **one shared collection that declares 12 vector fields**:

- 10 `SPARSE_FLOAT_VECTOR` fields created in the loop `bad_metric_l2 … bad_block_size_negative` (lines 969-981)
- 2 more `SPARSE_FLOAT_VECTOR` BM25-output fields `bad_bm25_quant` / `bad_bm25_quant_literal` (lines 983-984)

The default `proxy.maxVectorFieldNum` is `10` (default since #47866), so `create_collection` is rejected at setup:

```
RPC error: [create_collection] ... maximum vector field's number should be limited to 10: invalid parameter
AssertionError: Response of API create_collection expect True, but got False
```

Because `prepare_collection` is a **class-scoped autouse fixture**, the setup failure makes **all 9 negative cases ERROR**:
`test_sparse_index_rejects_unsupported_metric`, `test_ip_rejects_bm25_quant_type`, `test_bm25_rejects_ip_quant_type`, `test_sparse_index_rejects_unknown_quant_type`, `test_sparse_index_rejects_invalid_algo`, `test_sindi_rejects_invalid_window_size`, `test_sparse_index_rejects_invalid_codec`.

### Why it was not caught before merge

The class is tagged `CaseLabel.L2` (line 954). The PR merge gate runs **L0/L1 only**, so these L2 `milvus_client/` cases never executed pre-merge. It surfaced when running the full **L0+L1+L2** `milvus_client/` suite locally.

### Expected Behavior

The negative-test collection must not exceed `maxVectorFieldNum`; the 9 cases should run and assert the index-param rejections they were written for.

### Steps To Reproduce

```bash
cd tests/python_client
pytest milvus_client/test_milvus_client_sparse_inverted_index.py::TestSparseInvertedIndexV3Negative -p no:xdist
```

### Suggested Fix

Split the negative schema across ≥2 collections (each ≤10 vector fields), or reuse fewer fields (drop+recreate index per case). Prefer this over raising `proxy.maxVectorFieldNum` globally in the e2e config just for this suite.

### Related

- Test added in #50105 (cc @FeilongHou)
- Tangential feature: #42824 (support a larger vector-field limit)
- The same L2 `milvus_client/` sweep also needs pymilvus ≥ rc9 for binary struct-array element search (`test_element_filter_search_binary`, client raises `unsupported data type: 106` on rc8) — already covered by the pymilvus-bot bump PR #50249 (→ `3.1.0rc25`); verified passing on rc25.
- Broader gap: L2 `milvus_client/` cases are not exercised by the PR merge gate, so test-only breakages like this land unnoticed.

--- a/tests/python_client/milvus_client/test_milvus_client_sparse_inverted_index.py
+++ b/tests/python_client/milvus_client/test_milvus_client_sparse_inverted_index.py
@@ -960,6 +960,7 @@ class TestSparseInvertedIndexV3Negative(_SparseInvertedIndexV3Base):
     def setup_class(self):
         super().setup_class(self)
         self.collection_name = "TestSparseInvertedIndexV3Negative" + cf.gen_unique_str("sparse_v3_bad")
+        self.bm25_collection_name = "TestSparseInvertedIndexV3Negative" + cf.gen_unique_str("sparse_v3_bad_bm25")
 
     @pytest.fixture(scope="class", autouse=True)
     def prepare_collection(self, request):
@@ -979,10 +980,14 @@ class TestSparseInvertedIndexV3Negative(_SparseInvertedIndexV3Base):
             "bad_block_size_negative",
         ]:
             schema.add_field(field_name, DataType.SPARSE_FLOAT_VECTOR)
-        schema.add_field("bad_text", DataType.VARCHAR, max_length=256, enable_analyzer=True)
-        schema.add_field("bad_bm25_quant", DataType.SPARSE_FLOAT_VECTOR)
-        schema.add_field("bad_bm25_quant_literal", DataType.SPARSE_FLOAT_VECTOR)
-        schema.add_function(
+        self.create_collection(client, self.collection_name, schema=schema, force_teardown=False)
+
+        bm25_schema = client.create_schema(auto_id=False, enable_dynamic_field=False)
+        bm25_schema.add_field("id", DataType.INT64, is_primary=True)
+        bm25_schema.add_field("bad_text", DataType.VARCHAR, max_length=256, enable_analyzer=True)
+        bm25_schema.add_field("bad_bm25_quant", DataType.SPARSE_FLOAT_VECTOR)
+        bm25_schema.add_field("bad_bm25_quant_literal", DataType.SPARSE_FLOAT_VECTOR)
+        bm25_schema.add_function(
             Function(
                 name="bad_bm25_function",
                 function_type=FunctionType.BM25,
@@ -991,7 +996,7 @@ class TestSparseInvertedIndexV3Negative(_SparseInvertedIndexV3Base):
                 params={},
             )
         )
-        schema.add_function(
+        bm25_schema.add_function(
             Function(
                 name="bad_bm25_literal_function",
                 function_type=FunctionType.BM25,
@@ -1000,15 +1005,23 @@ class TestSparseInvertedIndexV3Negative(_SparseInvertedIndexV3Base):
                 params={},
             )
         )
-        self.create_collection(client, self.collection_name, schema=schema, force_teardown=False)
+        self.create_collection(client, self.bm25_collection_name, schema=bm25_schema, force_teardown=False)
 
         def teardown():
-            self.drop_collection(self._client(alias=self.shared_alias), self.collection_name)
+            client = self._client(alias=self.shared_alias)
+            self.drop_collection(client, self.collection_name)
+            self.drop_collection(client, self.bm25_collection_name)
 
         request.addfinalizer(teardown)
 
+    def _collection_for_field(self, field_name):
+        if field_name.startswith("bad_bm25"):
+            return self.bm25_collection_name
+        return self.collection_name
+
     def _assert_create_index_fails(self, field_name, metric_type, params, error_message):
         client = self._client(alias=self.shared_alias)
+        collection_name = self._collection_for_field(field_name)
         index_params = self.prepare_index_params(client)[0]
         index_params.add_index(
             field_name=field_name,
@@ -1018,12 +1031,12 @@ class TestSparseInvertedIndexV3Negative(_SparseInvertedIndexV3Base):
         )
         self.create_index(
             client,
-            self.collection_name,
+            collection_name,
             index_params=index_params,
             check_task=CheckTasks.err_res,
             check_items={ct.err_code: 1100, ct.err_msg: error_message},
         )
-        indexes, _ = self.list_indexes(client, self.collection_name, field_name=field_name)
+        indexes, _ = self.list_indexes(client, collection_name, field_name=field_name)
         assert indexes == []
 
     @pytest.mark.parametrize(

